cli/src/test/java/com/devonfw/tools/ide/io/FileAccessImplTest.java

```diff
@@ -267,6 +270,52 @@ public void testSymlinkShortcutPaths(@TempDir Path tempDir) {
     }
   }
 
+  /**
+   * Test of {@link FileAccessImpl#download(String, Path)} with default value for missing content length if the url starts with http.
+   */
+  @Test
+  public void testDownloadWithDefaultValueForMissingContentLength() {
+
+    // arrange
+    IdeTestContext context = newContext("basic");
+    FileAccess fileAccess = new FileAccessImpl(context);
+    String url = "https://getsamplefiles.com/download/zip/sample-1.zip";

```

Comment:
Aint it a bug of this external service to have no `Content-Length` header?
Assuming they will "fix" this "bug" then this test will break? Or not break anymore even in case the code is not working without `Content-Length`?
If we need to test something like this explicitly then probably wiremock is the better choice and for the progress bar we already have tests using wiremock...